src/subscriptions.ts

```diff
@@ -671,34 +671,48 @@ function getFilteredIndexerTransactions(
   transaction: algosdk.indexerModels.Transaction,
   filter: NamedTransactionFilter,
 ): SubscribedTransaction[] {
+  let rootOffset = 0
+  const getRootOffset = () => rootOffset++
+
   let parentOffset = 0
   const getParentOffset = () => parentOffset++
 
-  const transactions = [
-    new SubscribedTransaction({ ...transaction, filtersMatched: [filter.name] }),
-    ...getIndexerInnerTransactions(transaction, transaction, getParentOffset),
-  ] satisfies SubscribedTransaction[]
+  const innerTransactions = getIndexerInnerTransactions(transaction, transaction, transaction.id!, getRootOffset, getParentOffset)
+  const rootTransaction = new SubscribedTransaction({ ...transaction, innerTxns: innerTransactions, filtersMatched: [filter.name] })
```

Comment:
do we need the `filtersMatched` thing on the inner transactions?

Reply:
(no? because the one that is matched is this top level one? In which case this top level one may not be a root transaction in the block? In which case its intraRoundOffset may be wrong initially?)